csv-core/src/reader.rs

```diff
@@ -913,106 +926,105 @@ impl Reader {
         }
     }
 
+
     /// Compute the next NFA state given the current NFA state and the current
     /// input byte.
     ///
-    /// This returns the next NFA state along with two booleans that indicate
-    /// whether an input byte was consumed (i.e., a non-epsilon transition)
-    /// and whether the input byte should be copied to a caller provided output
-    /// buffer. For example, seeing a field delimiter will consume an input
-    /// byte but should not copy it to the output buffer.
+    /// This returns the next NFA state along with an NfaInputAction that
+    /// indicates what should be done with the input byte (nothing for an epsilon
+    /// transition, copied to a caller provided output buffer, or discarded).
     ///
-    /// It is illegal for this to emit an output byte without consuming the
-    /// input byte.
+    /// Note that this is more like a DFA than a true NFA; you cannot transition
+    /// to multiple states.
```

Comment:
> But I was expecting to see something dealing with the power set of possible states.

Indeed. It's there, in `build_dfa`, but is probably not what you're expecting. Since the alphabet and the set of states is so small, it simply runs the NFA through every possible set of inputs. The epsilon closure is computed by virtue of running the NFA for each input byte until it reaches a non-epsilon transition. For example, the DFA state `StartRecord` includes the following NFA states: `StartRecord`, `StartField`, `EndFieldTerm`, `InRecordTerm`. The implementation somewhat obscures this.